storescu/src/main.rs

```diff
@@ -39,183 +41,215 @@ struct App {
-    };
-
-    let ts = TransferSyntaxRegistry
-        .get(&pc_selected.transfer_syntax)
-        .ok_or("Poorly negotiated transfer syntax")?;
-
     if verbose {
-        println!("Transfer Syntax: {}", ts.name());
+        println!("Establishing association with '{}'...", &addr);
     }
 
-    let cmd = store_req_command(
-        storage_sop_class_uid,
-        storage_sop_instance_uid,
-        message_id,
-    );
-
-    let mut cmd_data = Vec::with_capacity(128);
-    cmd.write_dataset_with_ts(
-        &mut cmd_data,
-        &dicom::transfer_syntax::entries::IMPLICIT_VR_LITTLE_ENDIAN.erased(),
-    )?;
-
-    let mut object_data = Vec::with_capacity(2048);
-    dicom_file.write_dataset_with_ts(&mut object_data, ts)?;
-
-    let nbytes = cmd_data.len() + object_data.len();
+    let mut scu_init = ClientAssociationOptions::new();
+    
+    for (storage_sop_class_uid,transfer_syntax) in &presentation_contexts {
+        scu_init = scu_init.with_abstract_syntax(storage_sop_class_uid)
+                    .with_transfer_syntax(transfer_syntax);
+    }
+    let mut scu = scu_init.calling_ae_title(calling_ae_title)
+                    .called_ae_title(called_ae_title)
+                    .max_pdu_length(max_pdu_length)
+                    .establish(addr)?;
 
     if verbose {
-        println!("Sending payload (~ {} kB)...", nbytes / 1_000);
+        println!("Association established");
     }
 
-    if nbytes < scu.acceptor_max_pdu_length() as usize - 100 {
-        let pdu = Pdu::PData {
-            data: vec![
-                PDataValue {
-                    presentation_context_id: pc_selected.id,
-                    value_type: PDataValueType::Command,
-                    is_last: true,
-                    data: cmd_data,
-                },
-                PDataValue {
-                    presentation_context_id: pc_selected.id,
-                    value_type: PDataValueType::Data,
-                    is_last: true,
-                    data: object_data,
-                },
-            ],
-        };
-
-        scu.send(&pdu)?;
-    } else {
-        let pdu = Pdu::PData {
-            data: vec![PDataValue {
-                presentation_context_id: pc_selected.id,
-                value_type: PDataValueType::Command,
-                is_last: true,
-                data: cmd_data,
-            }],
-        };
-
-        scu.send(&pdu)?;
-
-        {
-            let mut pdata = scu.send_pdata(pc_selected.id);
-            pdata.write_all(&object_data)?;
+    for mut file in &mut dicom_files {
+        // TODO(#106) transfer syntax conversion is currently not supported
+        match check_presentation_contexts(file, scu.presentation_contexts()) {
+            Ok((pc, ts)) => {
+                file.pc_selected = Some(pc);
+                file.ts_selected = Some(ts);
+            },
+            Err(e) => {
+                if fail_first {
+                    eprintln!("Could not choose a transfer syntax: {}", e);
```

Comment:
```suggestion
                eprintln!("Could not choose a transfer syntax: {}", e);
                if fail_first {
```
The error should still be reported to the user.